### PR TITLE
Fix #138 - added functionality to get any custom attribute from an assembly.

### LINKED ISSUE #138: Function fileversioninfo::get-product-version should return string, not Version.
If `fileversioninfo::get-product-version(fileVersionInfo)` does what I think it does - return the value set by the `AssemblyInformationalVersion` attribute, it should return a `string` - not a `Version` instance. You can use the `AssemblyInformationalVersion` attribute to set a non-version-number supporting value on the assembly. I'd like to get this value back using Nant but it is upset because my value is not a version, but a plain string.

Version 0.92.4543.0

### PR DESCRIPTION
**Pull request by JoshMcCullough**
I believe this would be helpful, in general.

### CODE REVIEW

**Comment by JoshMcCullough** [comment]
First commit adds additional functionality - take it or leave it. Second commit actually fixes the underlying issue with the `file-version-info::get-product-version` function.

**Comment by JoshMcCullough** [comment]
Any movement on getting this merged into master?

**Comment by rmboggs** [comment]
Hi,

I'm not sure I am comfortable with changes like this (changing existing function to return different value type).  I think it would be better to just create a new one called something like ```file-version-info::get-product-version-string``` and keep the existing one to avoid possible breakage with other user scripts.

Thanks,
Ryan

File: src/NAnt.Core/Functions/FileVersionInfoFunctions.cs [hunk 1]
```diff
@@ -101,15 +101,14 @@ public class FileVersionInfoFunctions : FunctionSetBase {
...
+        /// <param name="fileVersionInfo">A <see cref="string" /> containing version information about a file.</param>
         /// <returns>
         /// The product version of a file.
         /// </returns>
         /// <see cref="VersionFunctions" />
         [Function("get-product-version")]
-        public static Version GetProductVersion(FileVersionInfo fileVersionInfo) {
-            return new Version(fileVersionInfo.ProductMajorPart, fileVersionInfo.ProductMinorPart,
-                fileVersionInfo.ProductBuildPart, fileVersionInfo.ProductPrivatePart);
+        public static string GetProductVersion(FileVersionInfo fileVersionInfo) {
```
Comment (rmboggs) [inline src/NAnt.Core/Functions/FileVersionInfoFunctions.cs]: don't change the return type.  Just add a new function called ```public static string GetProductVersionString(FileVersionInfo fileVersionInfo)```...

**Comment by JoshMcCullough** [comment]
Fine by me. I can create a new pull request which doesn't alter the existing functionality.

**Comment by rmboggs** [comment]
That would be better, yeah.

If you can add a test case or 2, that would be helpful as well.  If not possible, just say so and I will see about getting it in.

**Pull request closed without merging**